plugins/search/src/components/SearchBar/SearchBar.tsx

```diff
@@ -14,14 +14,61 @@
  * limitations under the License.
  */
 
-import React, { ChangeEvent, useEffect, useState } from 'react';
+import React, { ChangeEvent, useEffect, KeyboardEvent, useState } from 'react';
 import { useDebounce } from 'react-use';
 import { InputBase, InputAdornment, IconButton } from '@material-ui/core';
 import SearchIcon from '@material-ui/icons/Search';
 import ClearButton from '@material-ui/icons/Clear';
 
 import { useSearch } from '../SearchContext';
 
+type PresenterProps = {
+  value: string;
+  onChange: (e: ChangeEvent<HTMLInputElement>) => void;
+  handleClear?: () => void;
+  onKeyDown?: (e: KeyboardEvent<HTMLInputElement>) => void;
+  className?: string;
+  placeholder?: string;
+};
+
+export const SearchBarPresenter = ({
+  value,
+  onChange,
+  onKeyDown,
+  handleClear,
+  className,
+  placeholder,
+}: PresenterProps) => {
+  const endAdornment = (
+    <InputAdornment position="end">
+      <IconButton aria-label="Clear term" onClick={handleClear}>
+        <ClearButton />
+      </IconButton>
+    </InputAdornment>
+  );
+
+  return (
+    <InputBase
+      data-testid="search-bar-next"
+      fullWidth
+      placeholder={placeholder ?? 'Search in Backstage'}
+      value={value}
+      onChange={onChange}
+      inputProps={{ 'aria-label': 'Search term' }}
+      startAdornment={
+        <InputAdornment position="start">
+          <IconButton aria-label="Query term" disabled>
+            <SearchIcon />
+          </IconButton>
+        </InputAdornment>
+      }
+      {...(handleClear ? { endAdornment } : {})}
+      {...(className ? { className } : {})}
+      {...(onKeyDown ? { onKeyDown } : {})}
```

Comment:
just a hint, object spread (as opposed to array spread) is actually very type resistant - if you try to spread a non-iterable type it just passes over it silently. Therefore, it's perfectly fine to do

```ts
{...(onKeyDown && { onKeyDown })}
```

ie to skip the empty case entirely. A boolean false or an undefined etc will just be skipped happily with no ill effects. Some might say it's ugly to rely on that behavior of course ;)

Reply:
Ah, I love these kinds of comments. Thanks a lot for pointing that out, that's actually nicer. I'll stick to that way in the future :)

Fixed in 6be71f1.